dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/RequestItemRepository.java

```diff
@@ -287,19 +288,18 @@ public Class<RequestItemRest> getDomainClass() {
      * Generate a link back to DSpace, to act on a request.
      *
      * @param token identifies the request.
-     * @return URL to the item request API, with the token as request parameter
-     *          "token".
+     * @return URL to the item request API, with /request-a-copy/{token} as the last URL segments
      * @throws URISyntaxException passed through.
      * @throws MalformedURLException passed through.
      */
-    private String getLinkTokenEmail(String token)
+    public String getLinkTokenEmail(String token)
             throws URISyntaxException, MalformedURLException {
         final String base = configurationService.getProperty("dspace.ui.url");
-
-        URI link = new URIBuilder(base)
-                .setPathSegments("request-a-copy", token)
-                .build();
-
-        return link.toURL().toExternalForm();
+        URIBuilder uriBuilder = new URIBuilder(base);
+        // Add request-a-copy/token to the existing path (without breaking /sub/dir dspace URLs)
+        URI uri = uriBuilder.setPath(
+                (uriBuilder.getPath() == null ? "" : StringUtils.stripEnd(uriBuilder.getPath(), ""))
+                        + "/request-a-copy/" + token).build();
```

Comment:
Rather than doing this string-wise, how about `Collections.addAll(uriBuilder.getPath(), "request-a-copy", token)`?  UriBuilder knows how to get the separators right in all cases.

Reply:
@mwoodiupui thanks for the review.

In my testing, simple usage like `URI uri = uriBuilder.setPathSegments(uriBuilder.getPath(), "request-a-copy", token).build();` results in extra stringify array elements ([null], or [], or encoded slashes) no matter how stripping is handled in that first segment (which is either null/empty or the sub-path).

Not sure where `Collections.addAll` exactly fits in (did you mean the static method?), though we could prepare a list instead of string concatenation, is that what you were thinking?

Here is a non-concatenation approach that works. I'm happy to run with this instead:


```
        List<String> segments = new LinkedList<>();
        if (StringUtils.isNotBlank(uriBuilder.getPath())) {
            segments.add(StringUtils.strip(uriBuilder.getPath(), "/"));
        }
        segments.add("request-a-copy");
        segments.add(token);
        URI uri = uriBuilder.setPathSegments(segments).build();
```

(I know concatenation is viewed as 'bad', I just didn't see this as necessarily any more readable or performant for such a small string in my first go around, and I couldn't see any security issues with concat either:)

If there is a shorter way using Collections.addAll that I've missed, please let me know

Reply:
I think your suggestion should work.

I forgot that trailing slashes are an unavoidable problem with HTTP URL syntax.  Should've tested before suggesting.  I've made [a Gist](https://gist.github.com/mwoodiupui/1a308774c1030fc8deb9b7a416f13cc0) of the version of my approach that eventually worked with every base URL I've tried, in case anyone is interested.

Reply:
Thanks @mwoodiupui , I have pushed the changes as recommended. Is it looking better to you now?